proxy/src/main/java/com/wavefront/agent/listeners/otlp/OtlpProtobufUtils.java

```diff
@@ -225,23 +234,26 @@ private static String fromAnyValue(AnyValue anyValue) {
     } else if (anyValue.hasDoubleValue()) {
       return Double.toString(anyValue.getDoubleValue());
     } else if (anyValue.hasArrayValue()) {
-      // TODO
-      // Golang implementation: https://github.com/open-telemetry/opentelemetry-collector/blob/7ed3f75ef84d9e9d11b175a0859060f765faca0b/model/pdata/common.go#L381-L384
-      // ref: https://github.com/open-telemetry/opentelemetry-proto/blob/27a10cd70f63afdbddf460881969f9ad7ae4af5d/opentelemetry/proto/common/v1/common.proto
-      if (OTLP_DATA_LOGGER.isLoggable(Level.FINEST)) {
-        OTLP_DATA_LOGGER.severe("Encountered ArrayValue but cannot convert to String");
+      List<AnyValue> values = anyValue.getArrayValue().getValuesList();
+      StringBuilder sb = new StringBuilder();
+      sb.append('[');
+      for (AnyValue value : values) {
+        // recursive call fromAnyValue()
+        sb.append(fromAnyValue(value));
+        sb.append(", ");
       }
+      // remove the last ", " in the end
+      sb.deleteCharAt(sb.length() - 1);
+      sb.deleteCharAt(sb.length() - 1);
```

Comment:
-2 ;)

Reply:
This is gone since switching to `StringJoiner`.